src/components/sbb-toast/sbb-toast.tsx

```diff
@@ -251,45 +252,55 @@ export class SbbToast implements ComponentInterface {
     });
   }
 
-  public render(): JSX.Element {
-    return (
-      <Host
-        data-state={this._state}
-        data-has-icon={this._namedSlots['icon'] || !!this.iconName}
-        data-has-action={this._namedSlots['action'] || this.dismissible}
-      >
-        <div class="sbb-toast__overlay-container">
-          <div
-            class="sbb-toast"
-            role={this._role} // Firefox needs this to abilitate screen readers
-            onAnimationEnd={(event: AnimationEvent) => this._onToastAnimationEnd(event)}
-          >
-            <div class="sbb-toast__icon">
-              <slot name="icon">{this.iconName && <sbb-icon name={this.iconName} />}</slot>
-            </div>
-
-            <div class="sbb-toast__content" aria-live={this.politeness}>
-              <slot onSlotchange={(event) => this._onContentSlotChange(event)} />
-            </div>
-
-            <div class="sbb-toast__action">
-              <slot name="action" onSlotchange={(event) => this._onActionSlotChange(event)}>
-                {this.dismissible && (
-                  <sbb-button
+  protected override render(): TemplateResult {
+    // ## Host attributes ##
+    setAttribute(this, 'data-state', this._state);
+    setAttribute(this, 'data-has-icon', this._namedSlots['icon'] || !!this.iconName);
+    setAttribute(this, 'data-has-action', this._namedSlots['action'] || this.dismissible);
+    // ####
+
+    return html`
+      <div class="sbb-toast__overlay-container">
+        <div
+          class="sbb-toast"
+          ${/* Firefox needs 'role' to enable screen readers */ ''}
+          role=${this._role ?? nothing}
+          @animationend=${(event: AnimationEvent) => this._onToastAnimationEnd(event)}
```

Comment:
Nitpick (https://lit.dev/docs/templates/expressions/#event-listener-expressions)
```suggestion
          @animationend=${this._onToastAnimationEnd}
```

Reply:
It makes sense 👍 I just wanna be sure that the right context is bonded